File: acmicpc/A01644.py
from sys import stdin


def get_primes(n):
    if n < 5:
        return [i for i in range(n) if i in (2, 3)]
    n, c = n + 6 - n % 6, 2 - (n % 6 > 1)
    sieve = [True] * (n // 3)
    for i in range(1, int(n**0.5) // 3 + 1):
        if sieve[i]:
            d, s, j = (k := 1 | 3*i + 1) * 2, k * k, k * (k + 4 - 2 * (i & 1))
            sieve[s // 3::d] = [False] * ((n//6 - s//6 - 1) // k + 1)
            sieve[j // 3::d] = [False] * ((n//6 - j//6 - 1) // k + 1)
    return [2, 3] + [1 | 3 * i + 1 for i in range(1, n // 3 - c) if sieve[i]]


def two_pointer(n: int) -> int:
    if not (primes := get_primes(n + 1)):
        return 0
    length = len(primes)
    primes += [0]
    slow = fast = sum_ = ans = 0
    while fast <= length:
        if sum_ < n:
            sum_ += primes[fast]
            fast += 1
        else:
            ans += sum_ == n
            sum_ -= primes[slow]
            slow += 1
    return ans


def reader():
    n = int(stdin.readline())
    print(two_pointer(n))


if __name__ == '__main__':
    reader()

File: acmicpc/A01644.java
package acmicpc;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.Arrays;

public class A01644 {
    private static ArrayList<Integer> getPrimes(int n) {
        if (n < 2) return new ArrayList<Integer>();
        boolean[] sieve = new boolean[n + 1];
        Arrays.fill(sieve, true);
        for (int comp = 2; comp * comp <= n; comp++)
            if (sieve[comp])
                for (int i = comp * 2; i <= n; i += comp)
                    sieve[i] = false;
        return new ArrayList<Integer>() {{ 
            add(2);
            for (int i = 3; i <= n; i += 2) if (sieve[i]) add(i);
        }};
    }

    private static int twoPointer(int n) {
        if (n == 1) return 0;
        ArrayList<Integer> p = getPrimes(n);
        int slow = 0, fast = 0, sum = p.get(fast), ans = 0, len = p.size();
        while (slow <= fast) {
            if (sum < n) {
                if (++fast == len) break;
                sum += p.get(fast);
            } else {
                ans += sum == n ? 1 : 0;
                sum -= p.get(slow++);
            }
        }
        return ans;
    }

    public void reader() throws NumberFormatException, IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        System.out.println(twoPointer(Integer.parseInt(br.readLine())));
    }

    public static void main (String[] args) throws NumberFormatException, IOException {
        A01644 test = new A01644();
        test.reader();
    }
}
